File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/state/detail/tag/TagDetailUIState.kt
package dev.subfly.yabacore.state.detail.tag

import androidx.compose.runtime.Immutable
import dev.subfly.yabacore.model.ui.BookmarkUiModel
import dev.subfly.yabacore.model.ui.TagUiModel
import dev.subfly.yabacore.model.utils.BookmarkAppearance
import dev.subfly.yabacore.model.utils.CardImageSizing
import dev.subfly.yabacore.model.utils.SortOrderType
import dev.subfly.yabacore.model.utils.SortType
import kotlin.uuid.ExperimentalUuidApi
import kotlin.uuid.Uuid

@OptIn(ExperimentalUuidApi::class)
@Immutable
data class TagDetailUIState(
    val tag: TagUiModel? = null,
    val query: String = "",
    val bookmarks: List<BookmarkUiModel> = emptyList(),
    val isSelectionMode: Boolean = false,
    val selectedBookmarkIds: Set<Uuid> = emptySet(),
    val bookmarkAppearance: BookmarkAppearance = BookmarkAppearance.LIST,
    val cardImageSizing: CardImageSizing = CardImageSizing.SMALL,
    val sortType: SortType = SortType.EDITED_AT,
    val sortOrder: SortOrderType = SortOrderType.DESCENDING,
    val isLoading: Boolean = false,
)

File: Darwin/YABACore/State/Detail/Tag/TagDetailUIState.swift
//
//  TagDetailUIState.swift
//  YABACore
//

import Foundation

public struct TagDetailUIState: Sendable {
    public var tagId: String?
    public var query: String
    public var selectionMode: Bool
    public var selectedBookmarkIds: Set<String>
    public var sortType: YabaCoreSortType
    public var sortOrder: YabaCoreSortOrderType
    public var bookmarkAppearance: YabaCoreBookmarkAppearance
    public var cardImageSizing: YabaCoreCardImageSizing?

    public init(
        tagId: String? = nil,
        query: String = "",
        selectionMode: Bool = false,
        selectedBookmarkIds: Set<String> = [],
        sortType: YabaCoreSortType = .editedAt,
        sortOrder: YabaCoreSortOrderType = .descending,
        bookmarkAppearance: YabaCoreBookmarkAppearance = .list,
        cardImageSizing: YabaCoreCardImageSizing? = .small
    ) {
        self.tagId = tagId
        self.query = query
        self.selectionMode = selectionMode
        self.selectedBookmarkIds = selectedBookmarkIds
        self.sortType = sortType
        self.sortOrder = sortOrder
        self.bookmarkAppearance = bookmarkAppearance
        self.cardImageSizing = cardImageSizing
    }
}
